sematic/resolvers/log_streamer.py

```diff
@@ -137,15 +225,9 @@ def ingested_logs(
         The amount of time between uploads
     remote_prefix:
         The prefix for the remote storage location where ingested logs live
-    max_tail_bytes:
-        The maximum number of bytes of logs to print to the original stdout
-        when the context exits. To disable, set to <=0. Defaults to 8kb
     uploader:
         An optional override for uploading the log file.
     """
-    # we will eventually want to switch to use a pipe instead of file as the
-    # buffer the subprocess reads from:
-    # https://stackoverflow.com/questions/73821235/python-file-truncate-with-one-writer-and-one-reader?noredirect=1#comment130351264_73821235
     uploader = uploader if uploader is not None else _do_upload
```

Comment:
Does this need to be parametrized? Why not hard-code? (I know it's not related to this PR).

Reply:
it enables testing using a fake uploader. Mocks don't work in subprocesses, so you have to pass stuff like this explicitly. There's a comment about that in the test